forum/profiles/views.py

```diff
@@ -89,6 +91,67 @@ def list(self, request, *args, **kwargs):
         Handles GET requests for listing startup profiles.
         """
         return super().list(request, *args, **kwargs)
+    
+    @action(detail=False, 
+            methods=['get'], 
+            url_path='viewed', 
+            url_name='viewed-startups',
+            permission_classes=[IsInvestor])
+    def viewed_startups(self, request):
+        """Retrieve a list of recently viewed startups for the authenticated investor"""
+        investor = get_object_or_404(InvestorProfile, user=request.user)
+        viewed_startups = ViewedStratups.objects.filter(investor=investor).order_by('-viewed_at')
+        
+        page = self.paginate_queryset(viewed_startups)
+        if page is not None:
+            serializer = ViewedStratupReadSerializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+            
+        serializer = ViewedStratupReadSerializer(viewed_startups, many=True)
+        return Response(serializer.data)
+    
+    @action(detail=False,
+            methods=['delete'],
+            url_path='viewed/clear',
+            url_name='clear-viewed',
+            permission_classes=[IsInvestor])
+    def clear_viewed_history(self, request):
+        """Clear the investor's viewed startups history"""
+        investor = get_object_or_404(InvestorProfile, user=request.user)
+        ViewedStratups.objects.filter(investor=investor).delete()
+        return Response(
+            {'detail': 'Viewed startups history cleared successfully'},
+            status=status.HTTP_204_NO_CONTENT
+        )
+
+    @swagger_auto_schema(tags=['Startup Views'])
+    @action(detail=True,
+            methods=['post'],
+            url_path='view',
+            url_name='view-startup',
+            permission_classes=[IsInvestor])
+    def track_view(self, request, pk=None):
+        """Track that a startup was viewed by the current user"""
+        
+        startup = self.get_object()
+        investor = InvestorProfile.objects.get(user=request.user)
+        
+        data = {
+            'investor': investor.id,
+            'startup': startup.id
+        }
+        serializer = ViewedStratupCreateSerializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        
+        return Response({'detail': f'View recorded for {startup.company_name}'}, status=status.HTTP_200_OK)
```

Comment:
If the `track_view` method encounters an issue (e.g., the startup doesn't exist), it could raise unhandled exceptions. Add error handling to make it more robust.